libs/dal/src/repositories/execution-details/execution-details.repository.ts

```diff
@@ -1,18 +1,13 @@
 import { ExecutionDetailsStatusEnum } from '@novu/shared';
-import { ExecutionDetailsEntity } from './execution-details.entity';
-import { ExecutionDetails } from './execution-details.schema';
-import { BaseRepository, Omit } from '../base-repository';
-import { Document, FilterQuery } from 'mongoose';
-
-class PartialExecutionDetailsEntity extends Omit(ExecutionDetailsEntity, ['_environmentId', '_organizationId']) {}
 
-type EnforceEnvironmentQuery = FilterQuery<PartialExecutionDetailsEntity & Document> &
-  ({ _environmentId: string } | { _organizationId: string });
```

Comment:
Removed these hacks to force the `_environmentId` or `_organizationId`, because it breaks all the Mongoose types.
When the query is passed to the model function, the query fields are optional, but we force them to be required and that breaks types.

Reply:
We added it mainly to help us (and the community) from missing those fields. While working on the implementation I encountered a couple of those issues in the project that was missing environment fields.

Reply:
@djabarovgeorge @scopsy @p-fernandez I think I've found the solution here will update PR with a change soon ;) so the enforcement will stay